src/node/audio_buffer_source.rs

```diff
@@ -402,18 +432,33 @@ impl AudioProcessor for AudioBufferSourceRenderer {
         // 2. the duration has been reached.
         if scope.current_time >= stop_time || self.render_state.buffer_time_elapsed >= duration {
             output.make_silent(); // also converts to mono
+
+            // @note: we need this check because this is called a until the program
+            // ends, such as if the node was never removed from the graph
```

Comment:
An AudioProcessor is never dropped from the graph as long as the AudioNode object is still alive in the control thread. You could manually `drop` it from the example and see if that changes anything. The reason we keep it is because the control thread could still connect it to other nodes, and maybe even restart it (not allowed for buffer source node I think..)
In any case the check is needed because the cleanup process is not instant. It peels of 1 layer per render quantum. So if there were any nodes feeding into this one, they are cleaned first, then the next render call this node is dropped.